### PR TITLE
[4.0] Empty State for Banner clients and tracks

### PR DESCRIPTION
**Pull request by brianteeman**
Same as the other PRs

If you have never used the banner component - in order to test the tracks part of this PR and make sure its not displayed when there is tracking info you will need to create a banner AND set the tracking to yes on the second tab of the banner.

image

## After

image

image

### CODE REVIEW

**Comment by PhilETaylor** [comment]
suggest removing the Delete tracks button too - because if pressed you get an error 

Screenshot 2021-04-27 at 18 32 23

File: administrator/components/com_banners/tmpl/clients/emptystate.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\Layout\LayoutHelper;
+
+$displayData = array(
```
Comment (PhilETaylor) [inline administrator/components/com_banners/tmpl/clients/emptystate.php:15]: Apparently we are going to use Short array syntax `[  ]` in new files, despite the coding guidelines never mentioning it

File: administrator/components/com_banners/tmpl/clients/emptystate.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		'textPrefix' => 'COM_BANNERS_CLIENT',
+		'formURL'    => 'index.php?option=com_banners&view=clients',
+		'helpURL'    => 'https://docs.joomla.org/Special:MyLanguage/Help40:Banners:_Clients',
+		'icon'       => 'icon-bookmark banners',
+);
+$user        = Factory::getUser();
```
Comment (PhilETaylor) [inline administrator/components/com_banners/tmpl/clients/emptystate.php:21]: Please dont use deprecated function calls. use 
```suggestion
$user        = Factory::getApplication()->getIdentity();
```

File: administrator/components/com_banners/tmpl/tracks/emptystate.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Layout\LayoutHelper;
+
+$displayData = array(
```
Comment (PhilETaylor) [inline administrator/components/com_banners/tmpl/tracks/emptystate.php:14]: Short array syntax here too

File: administrator/language/en-GB/com_banners.ini [hunk 4]
```diff
@@ -36,6 +36,9 @@ COM_BANNERS_BEGIN_HINT="Begin date (yyyy-mm-dd)"
 COM_BANNERS_BEGIN_LABEL="Begin Date"
 COM_BANNERS_CANCEL="Cancel"
 COM_BANNERS_CLICK="Click"
+COM_BANNERS_CLIENT_EMPTYSTATE_BUTTON_ADD="Add your first client"
+COM_BANNERS_CLIENT_EMPTYSTATE_CONTENT="A Banner Client contains contact details and further information such as tracking preferences."
+COM_BANNERS_CLIENT_EMPTYSTATE_TITLE="No Clients have been created yet."
 COM_BANNERS_CLIENT_EDIT="Edit Client"
```
Comment (PhilETaylor) [inline administrator/language/en-GB/com_banners.ini:42]: need to move Edit Client to above the EmptyState keys (/me starts singing his abc's)

File: administrator/components/com_banners/tmpl/clients/emptystate.php (hunk 1 above)
Comment (Quy) [reply administrator/components/com_banners/tmpl/clients/emptystate.php:15]: ... and reduce indentation.

**Pull request closed without merging**